sdk/core/azure-core/src/main/java/com/azure/core/cryptography/IKeyEncryptionKey.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.cryptography;
+
+import reactor.core.publisher.Mono;
+
+
+/**
+ * A key which is used to encrypt, or wrap, another key.
+ */
+public interface IKeyEncryptionKey {
```

Comment:
Should this just be called `EncryptionKey`, and the interface below be called `EncryptionKeyResolver`?

Reply:
Updated, Introduced Sync and Async variants,

Reply:
Actually while it might seem verbose this should be `KeyEncryptionKey` and `KeyEncryptionKeyResolver`. The methods that `KeyEncryptionKey` exposes, `wrapKey` and `unwrapKey` are specific to encrypting a ___key___, not arbitrary data. An `EncryptionKey` interface would have methods `encrypt` and `decrypt`. 

Originally in track 1 there was the `IKey` interface which had methods,  `encrypt`, `decrypt`, `wrapKey`, `unwrapKey`, `sign`, and `verify`.  Here we have scoped an interface to only provide the methods needed by storage for their client side encryption implementation. But we want to name it appropriately so that we could reintroduce the full interface if necessary. The full set of interfaces would look something like this:

- KeyEncryptionKey 
  - wrapKey
  - unwrapKey
- EncryptionKey
  - encrypt
  - decrypt
- SignatureKey
  - sign
  - verify

Also you could possibly tie these all together in an interface to have an interface that was equivalent to track 1's `IKey`, although you'd need to rename it to fit the java guidelines, so something like `CryptographicKey` possibly.

Reply:
Your explanation makes sense. @g2vinay be sure to include some of what Scott has stated here in the class-level javadocs to clarify to users what this API is setting out to achieve.

@schaabs Because I'm no expert on crypto, my initial instinct is to have it all in one interface. My reasoning being is that I'd rather be overwhelmed in one interface rather than be overwhelmed by multiple interfaces! :-) However, for this I'm deferring to you, I just wanted to share a naive opinion on crypto API intuitiveness.

Reply:
updated names to KeyEncryptionKey and KeyEncryptionKeyResolver